### ISSUE TITLE
Ubuntu packages are really old

### ISSUE THREAD

**Issue by stuaxo** [direnv/direnv#951]
I realise this is a downstream issue, but does anyone have an idea who maintains the package for direnv at ubuntu so I can ask if they can update it ?

I use .env files for work, and their older version doesn't support this.

I am using the released version from github, but know I will forget this in future when setting up a computer.

**Comment by lofidevops** [comment]
If direnv resumed releasing to PyPI https://pypi.org/project/direnv/#history this could be resolved by using [pipx](https://pypa.github.io/pipx/) as a package manager (`pipx install direnv`).

**Comment by punitagrawal** [comment]
Hi @zimbatm, Thanks for looping me in.
You're right - in the majority of cases, Ubuntu reuses Debian packaging efforts.

With regards to direnv, the package update has fallen behind due to missing dependencies in Debian. The de-vendoring of libraries around v25.2 means that now those dependencies need to be maintained as separate packages in Debian. Technically it's a step in the right direction, but it is an overhead on the packaging front. Also, adding new packages have a delay due to additional checks / reviews that are required before being accepted into the archive.

It's on my todo list to get the ball rolling in this regard but I don't have an ETA yet. I need to sync up with the Go packaging team in Debian to see if there are tricks I am missing regarding the above.

**Comment by zimbatm** [comment]
I didn't realize it would cause so many problems to de-vendor. Is the second package just holding the modules source?

If that's an issue I don't mind re-vendoring them with `go mod vendor` this time.

**Comment by punitagrawal** [comment]
Sorry for the delay in getting back. I wanted to check a few things before replying. 

Based on the latest direnv release, building it requires the following external dependencies - 

github.com/direnv/go-dotenv
github.com/BurntSushi/toml
github.com/mattn/go-isatty
golang.org/x/mod
golang.org/x/sys

Of these, `go-dotenv` is the only dependency that's currently missing. Though some alternatives listed in the go-dotenv README.md (such as github.com/joho/godotenv, github.com/subosito/gotenv) are available. Not sure if they are drop in replacement.

> Is the second package just holding the modules source?

Indeed - the go library packages seem to hold the the module sources as dynamically linked libraries are not a thing in go. Not sure if things have changed in this regard.

> If that's an issue I don't mind re-vendoring them with `go mod vendor` this time.

If partial vendoring is a thing, bringing back go-dotenv (or using one of the available alternative) maybe the quickest way forward for updating the package in Debian.

Longer term, if separating go-dotenv is the right thing to do, it can be uploaded as a separate package and once available it can be dropped from direnv proper.

**Comment by stuaxo** [comment]
@punitagrawal indeed, if you can do partial vendoring, a ticket tracking the list of vendored tickets would be handy, they could be ticked off as they get packaged, until none are left.

**Comment by zimbatm** [comment]
Ok, I will move github.com/direnv/go-dotenv back to direnv for the next release

**Comment by zimbatm** [comment]
@punitagrawal here, for you :) https://github.com/direnv/direnv/releases/tag/v2.32.1

**Comment by zimbatm** [comment]
sweet, let me know when it's done and I will close this issue

**Comment by punitagrawal** [comment]
Apologies for the delay!

I have now updated the direnv package in `unstable` to 2.32.1[0][1]. Barring any problems, it should transition to `testing` in a few days. Ubuntu will also include the new version in their next release - which is the issue reported here.

[0] https://tracker.debian.org/news/1346786/accepted-direnv-2321-1-source-into-unstable/
[1] https://tracker.debian.org/pkg/direnv

**Comment by zimbatm** [comment]
Makes sense. I wanted to avoid the PRs. This time I disabled dependabot in the repo setting instead, let's see if that works better.

**Comment by Ativerc** [comment]
Any updates on this?

**Comment by Ativerc** [comment]
Ah, I missed that comment. My apologies.

**Comment by punitagrawal** [comment]
> is there any recourse for users of old Ubuntu versions (I'm on Ubuntu 20.04)?

In Debian, some packages tend to be backported and can be accessed by enabling the backports archive. Not sure if Ubuntu provides backports. Or upgrading to a newer version but I guess this is not an option.

Personally, I tend to use latest releases for development machines with chroots / containers for old environment as needed. But I understand this is not always feasible.
